README.md

```diff
@@ -325,6 +325,36 @@ To reimport just one entity, run:
 bundle exec rails dfe:analytics:import_entity[entity_name]
 ```
 
+
+## Event debugging
+
+If you wish to log events for debug purposes, create a file `config/analytics_event_debug.yml` containing an array of your event filters  under a `shared` key like:
+
+```yaml
+shared:
+  event_filters:
+    -
+      type: (create|update|delete)_entity
+      entity_table_name: course_options
+      key: id
+      value: 12345
+    -
+      type: update_entity
+      entity_table_name: courses
+```
+
+Event filters allow targeted event logging for diagnostic and debug purposes. The logging level is `info`.
+
+When defining event filters, note the following:
+- All values are converted to regular expressions for matching
+- Any filter fields can be defined as long as the field exists in the target event
+- A filter must be a hash and nested fields are allowed
+- If a corresponding hash field in the target event is not found, then the remaining value in the target is converted into a string and compared with the value from the filter. The remaining nested fields in the filter are then ignored. This may result in a wider match than expected
```

Comment:
I'm struggling to understand this condition. Could we simplify somehow? (perhaps by removing this behaviour?)

Reply:
So this is probably best demonstrated by example. So given the example filter:

``` yaml
shared:
  event_filters:
    -
      event_type: (create|update|delete)_entity
      entity_table_name: course_options
      data:
        key: id
        value: 12345
```

Also the target event:

``` Ruby
  {
    'entity_table_name' => 'course_options',
    'event_type' => 'update_entity',
    'data' => [
      { 'key' => 'id', 'value' => ['12345'] },
      { 'key' => 'course_id', 'value' => ['42'] }
    ]
  }
```

Then on matching, there is a one to one correspondence on the `entity_table_name` and `event_type` fields, so thees matches are OK.
However, in the `data` field there is no hash value, so the `key` field with value of `id` is compared with the whole of the target data field converted to a string, and the `value` field with value of `['12345']` would also be compared.

So the comparisons would be:

``` Ruby
  /id/ =~ "[{ 'key' => 'id', 'value' => ['12345'] }, { 'key' => 'course_id', 'value' => ['42'] }]"
  /\['12345'\]/ =~ "[{ 'key' => 'id', 'value' => ['12345'] }, { 'key' => 'course_id', 'value' => ['42'] }]"

```

Matching nested fields in a hash generically is going to be non-trivial, and I'm struggling to explain this in a simple fashion, but as you point out this is not necessarily required for the problem in hand and maybe best to keep it simple.